lib/dao.js

```diff
@@ -167,9 +168,9 @@ DataAccessObject.create = function (data, options, cb) {
     }
   }
 
-  cb = cb || noCallback;
   data = data || {};
   options = options || {};
+  cb = cb || (Array.isArray(data) ? noCallback : utils.createPromiseCallback());
```

Comment:
I did it this way to preserve the existing behavior, returning data when called with an array and no callback. 
Probably best not to break any existing API behavior, at least in this PR...